### ISSUE TITLE
lint() output should be class c("lints", "list") for S3 dispatch

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1494]
I am working on something where I need to edit the line numbers of the `lint()` output; a natural way to do that is like so:

```r
l <- lint(...)
l <- lapply(l, within, line_number <- line_number + 1)
```

But this fails because `within` attempts to dispatch on `lints` class. For now, we need to call the method explicitly:

```r
l <- lapply(l, within.list, line_number <- line_number + 1)
```

We could also define a `within.lints()` method, but that only works for this specific case, I think the problem is more general (it will affect other generics used too process the `lint()` output, too).

**Comment by AshesITR** [comment]
Ditto for `Lint()`, right?
